pkg/provisioner/crdprovisioner.go

```diff
@@ -546,7 +546,7 @@ func (c *CrdProvisioner) PublishVolume(
 	filterVAToDetach := func(attachments []azdiskv1beta2.AzVolumeAttachment) (numAttachment int, unpublishOrder []*azdiskv1beta2.AzVolumeAttachment) {
 		for i := range attachments {
 			// only count if deletionTimestamp not set
-			if attachments[i].GetDeletionTimestamp().IsZero() {
+			if attachments[i].GetDeletionTimestamp().IsZero() && !azureutils.MapContains(attachments[i].Status.Annotations, consts.VolumeDetachRequestAnnotation) {
```

Comment:
`volumeDetachRequested` should only check for `VolumeDetachRequestAnnotation` because we now would be removing finalizer from the object when `deletionTimestamp` is set.